python_modules/dagit/bin/dagit

```diff
@@ -1,19 +1,41 @@
-#!/bin/bash
-ARGS=()
-WATCH="0"
-for var in "$@"; do
-    if [ "$var" == '--no-watch' ]; then
-      WATCH="0"
-    if [ "$var" == '--watch' ]; then
-      WATCH="1"
-    else
-      ARGS+=("$var")
-    fi
-done
+#!/usr/bin/env python
+import sys
+import time
+import signal
+from watchdog.observers import Observer
+from watchdog.tricks import AutoRestartTrick
 
-if [ "$WATCH" = "1" ]; then
-  echo "Watching for file changes..."
-  watchmedo auto-restart -p="*.py" -R dagit-cli -- "$@"
-else
-  dagit-cli "${ARGS[@]}"
-fi
+# Build the dagit-cli command, omitting the --no-watch arg if present
+watch = True
+command = ["dagit-cli"]
+for arg in sys.argv[1:]:
+    if arg == "--no-watch":
+        watch = False
+    else:
+        command.append(arg)
+
+handler = AutoRestartTrick(command=command,
+                           patterns=["*.py"],
+                           ignore_patterns=[],
+                           ignore_directories=[],
+                           stop_signal=signal.SIGINT,
+                           kill_after=0.5)
+handler.start()
+
+if watch:
+    print("Watching for file changes...")
+    observer = Observer(timeout=1)
```

Comment:
do timeout=1 and sleep(1) have to be the same number?

Reply:
So this is interesting actually - it looks like this timeout is the `Read events blocking timeout / polling interval (in fallback scenarios)` and it defaults to 1, and their source also included a sleep of 1s in the other location. I think that setting this timeout to a value lower than the sleep value may have no effect, but it's hard to tell. Either way these seem to be the defaults they're using in watchmedo so should be OK for now.